core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -669,6 +686,31 @@ private[spark] class AppStatusListener(
         }
       }
     }
+
+    // check if there is a new peak value for any of the executor level memory metrics
+    // for the live UI. SparkListenerExecutorMetricsUpdate events are only processed
+    // for the live UI.
+    event.executorUpdates.foreach { updates: ExecutorMetrics =>
```

Comment:
Does the type need to be explicitly defined here and in the next line?

Reply:
No, these don't need to be explicitly defined.